src/natcap/invest/stormwater.py

```diff
@@ -0,0 +1,939 @@
+        2D numpy array of adjusted retention ratios. Has the same shape as
+        ``retention_ratio_array``.
+    """
+    adjusted_ratio_array = numpy.full(ratio_array.shape, FLOAT_NODATA,
+                                      dtype=numpy.float32)
+    adjustment_factor_array = numpy.full(ratio_array.shape, FLOAT_NODATA,
+                                         dtype=numpy.float32)
+    valid_mask = (
+        ~numpy.isclose(ratio_array, FLOAT_NODATA) &
+        ~numpy.isclose(avg_ratio_array, FLOAT_NODATA) &
+        (near_impervious_lulc_array != UINT8_NODATA) &
+        (near_road_array != UINT8_NODATA))
+
+    # adjustment factor:
+    # - 0 if any of the nearby pixels are impervious/connected;
+    # - average of nearby pixels, otherwise
+    is_not_impervious = ~(
+        near_impervious_lulc_array[valid_mask] |
+        near_road_array[valid_mask]).astype(bool)
+    adjustment_factor_array[valid_mask] = (avg_ratio_array[valid_mask] *
+                                           is_not_impervious)
+
+    adjustment_factor_array[valid_mask] = (
+        avg_ratio_array[valid_mask] * ~(
+            near_impervious_lulc_array[valid_mask] |
+            near_road_array[valid_mask]
+        ).astype(bool))
+
+    # equation 2-4: Radj_ij = R_ij + (1 - R_ij) * C_ij
+    adjusted_ratio_array[valid_mask] = (ratio_array[valid_mask] +
+                                        (1 - ratio_array[valid_mask]) * adjustment_factor_array[valid_mask])
+    return adjusted_ratio_array
+
+
+def aggregate_results(base_aggregate_areas_path, target_path, wkt, aggregations):
+    """Aggregate outputs into regions of interest.
+
+    Args:
+        base_aggregate_areas_path (str): path to vector of polygon(s) to
+            aggregate over. This is the original input.
+        target_path (str): path to write out the results. This will be a copy
+            of the base vector with added fields, reprojected to the target WKT
+            and saved in geopackage format.
+        wkt (str): a Well-Known Text representation of the target spatial
+            reference. The base vector is reprojected to this spatial reference
+            before aggregating the rasters over it.
+        aggregations (list[tuple(str,str,str)]): list of tuples describing the
+            datasets to aggregate. Each tuple has 3 items. The first is the
+            path to a raster to aggregate. The second is the field name for
+            this aggregated data in the output vector. The third is either
+            'mean' or 'sum' indicating the aggregation to perform.
+
+    Returns:
+        None
+    """
+    pygeoprocessing.reproject_vector(base_aggregate_areas_path, wkt,
+                                     target_path, driver_name='GPKG')
+    # create a copy of the aggregate areas vector to write to
+    aggregate_vector = gdal.OpenEx(target_path, gdal.GA_Update)
+    aggregate_layer = aggregate_vector.GetLayer()
+
+    for raster_path, field_id, op in aggregations:
```

Comment:
This is minor, but might I suggest `aggregate_op` or something similar to help add a bit of meaning to `op`?

Reply:
Sure, changed it to `aggregation_op`